src/Microsoft.AspNet.OData.Shared/Query/ParameterAliasNodeTranslator.cs

```diff
@@ -398,7 +398,7 @@ public override QueryNode Visit(UnaryOperatorNode nodeIn)
         /// <returns>The translated node.</returns> 
         public override QueryNode Visit(CountNode nodeIn)
         {
-            return new CountNode((CollectionNode)nodeIn.Source.Accept(this));
+            return new CountNode((CollectionNode)nodeIn.Source.Accept(this), nodeIn.FilterClause, nodeIn.SearchClause);
```

Comment:
Did you cater for $search as well? If yes, maybe you should mention that in the PR description

Reply:
$search is not supported in WebAPI.
This code ensures we return the correct node after adding Parameter alias.